### PR TITLE
ceph: add pool quota usage alerts

### PR DESCRIPTION
**Pull request by umangachapagain**
Signed-off-by: Umanga Chapagain <chapagainumanga@gmail.com>

**Description of your changes:**
add quota usage alerts for pools with quota set.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [x] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [x] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[skip ci]

### CODE REVIEW

File: cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml [hunk 1]
```diff
@@ -326,3 +326,27 @@ spec:
...
+        severity_level: warning
+        storage_type: ceph
+      expr: |
+        ceph_pool_stored_raw / (ceph_pool_quota_bytes > 0) > 0.70
+      for: 1m
+      labels:
+        severity: warning
+    - alert: PoolQuotaBytesCriticallyExhausted
+      annotations:
+        description: Storage pool quota usage has crossed 90%.
```
Comment (anmolsachan) [inline cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:344]:
```suggestion
        description: Storage pool  {{ $labels.name }} quota usage has crossed 90%.
```

File: cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml [hunk 2]
```diff
@@ -326,3 +326,27 @@ spec:
       for: 0s
       labels:
         severity: critical
+  - name: pool-quota.rules
+    rules:
+    - alert: PoolQuotaBytesNearExhaustion
+      annotations:
+        description: Storage pool quota usage has crossed 70%.
+        message: Storage pool quota bytes is near exhaustion.
```
Comment (anmolsachan) [inline cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:334]:
```suggestion
        message: Storage pool quota(bytes) is near exhaustion.
```

File: cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml [hunk 3]
```diff
@@ -326,3 +326,27 @@ spec:
...
+        storage_type: ceph
+      expr: |
+        ceph_pool_stored_raw / (ceph_pool_quota_bytes > 0) > 0.70
+      for: 1m
+      labels:
+        severity: warning
+    - alert: PoolQuotaBytesCriticallyExhausted
+      annotations:
+        description: Storage pool quota usage has crossed 90%.
+        message: Storage pool quota bytes is critically exhausted.
```
Comment (anmolsachan) [inline cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:345]:
```suggestion
        message: Storage pool quota(bytes) is critically exhausted.
```

**Review by anmolsachan — CHANGES_REQUESTED** [review]

File: cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml [hunk 4]
```diff
@@ -326,3 +326,27 @@ spec:
       for: 0s
       labels:
         severity: critical
+  - name: pool-quota.rules
+    rules:
+    - alert: PoolQuotaBytesNearExhaustion
```
Comment (anmolsachan) [inline cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:331]: Considering support of other Storage Providers, we can add Ceph to the name of the alert.
```suggestion
    - alert: CephPoolQuotaBytesNearExhaustion
```

File: cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml [hunk 5]
```diff
@@ -326,3 +326,27 @@ spec:
...
+        description: Storage pool {{ $labels.name }} quota usage has crossed 70%.
+        message: Storage pool quota(bytes) is near exhaustion.
+        severity_level: warning
+        storage_type: ceph
+      expr: |
+        (ceph_pool_stored_raw * on (pool_id) group_left(name)ceph_pool_metadata) / ((ceph_pool_quota_bytes * on (pool_id) group_left(name)ceph_pool_metadata) > 0) > 0.70
+      for: 1m
+      labels:
+        severity: warning
+    - alert: PoolQuotaBytesCriticallyExhausted
```
Comment (anmolsachan) [inline cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:342]: Considering the support of other Storage Providers, we can add Ceph to the name of the alert.
```suggestion
    - alert: CephPoolQuotaBytesCriticallyExhausted
```

**Review by anmolsachan — CHANGES_REQUESTED** [review]

**Review by anmolsachan — APPROVED** [review]

**Review by travisn — APPROVED** [review]

**Comment by zerkms** [comment]
@umangachapagain @travisn  I think rules added in this PR are wrong, see https://github.com/rook/rook/issues/8735

**Pull request merged**